src/reactapp/src/components/couponCode/components/CouponCodeForm.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react';
+
+import Button from '../../common/Button';
+import TextInput from '../../common/Form/TextInput';
+import { __ } from '../../../i18n';
+import {
+  useCouponCodeAppContext,
+  useCouponCodeFormContext,
+  useCouponCodeCartContext,
+} from '../hooks';
+
+function CouponCodeForm() {
+  const [codeChecked, setCodeChecked] = useState('');
+  const { fields, formikData, setFieldError, setFieldTouched, setFieldValue } =
+    useCouponCodeFormContext();
+  const { setPageLoader, setSuccessMessage, setErrorMessage } =
+    useCouponCodeAppContext();
+  const { appliedCoupon, applyCouponCode, removeCouponCode } =
+    useCouponCodeCartContext();
+  const couponCode = formikData?.formSectionValues?.couponCode;
+
+  const handleApplyCoupon = async (code) => {
+    try {
+      setPageLoader(true);
+      await applyCouponCode(code);
+      setCodeChecked(code);
+      setSuccessMessage(
+        __('Coupon code: %1 is applied successfully.', couponCode)
```

Comment:
@rajeev-k-tomy Do we already have the translations for these new strings?

Reply:
Nope, translations need to be incorporated.

Reply:
> Just a question regarding translations, otherwise looks very good!

I am merging this as the layout xml file changes happened from other PR. So the translations would be applied in a separate PR.